src/puppeteer/puppeteer-plugin.ts

```diff
@@ -21,6 +23,22 @@ export class PuppeteerPlugin extends BrowserPlugin<typeof Puppeteer> {
         };
 
         let browser = await this.library.launch(finalLaunchOptions);
+
+        browser.on('targetcreated', async (target: Puppeteer.Target) => {
+            try {
+                const page = await target.page();
+
+                if (page) {
+                    page.on('error', (error) => {
+                        log.exception(error, 'Page crashed.');
+                        page.close().catch(noop);
```

Comment:
I guess there's no way to avoid this race condition? There is "some" short time between this handler is attached and the page being created. I hope that this attaches faster than the page can crash.

Reply:
```js
const puppeteer = require('puppeteer');

(async () => {
    const browser = await puppeteer.launch();

    browser.on('targetcreated', async target => {
        await target.page();
        console.log('got target page');
    });

    await browser.newPage();
    console.log('got page');

    await browser.close();
})();
```

```
got target page
got page
```

I think we have to just believe it'll be like that. I see no other easy way right now.

Reply:
> I hope that this attaches faster than the page can crash.

I don't know much about how this works, but what page gets opened to get the browser version? Maybe we could just open something guaranteed not to crash (e.g. some static local file)?

Reply:
> I don't know much about how this works, but what page gets opened to get the browser version?

No page. This is puppeteer, you can do `browser.version()`. If you're talking about playwright then it's just `about:blank` I guess.